debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/stackwalker/StackWalker.java

```diff
@@ -139,116 +154,165 @@ private static IStackWalker getImpl()
+				if (walkState.arg0EA == null) {
+					walkState.arg0EA = walkThread.arg0EA();
+				}
+				if (walkState.pc == null) {
+					walkState.pc = walkThread.pc();
+				}
+				walkState.pcAddress = walkThread.pcEA();
+				if (walkState.walkSP == null) {
+					walkState.walkSP = walkThread.sp();
+				}
+				if (walkState.literals == null) {
+					walkState.literals = walkThread.literals();
+				}
+				if (walkState.walkedEntryLocalStorage == null) {
+					walkState.walkedEntryLocalStorage = walkThread.entryLocalStorage();
+				}
+				walkState.j2iFrame = walkThread.j2iFrame();
+				walkState.decompilationStack = walkThread.decompilationStack();
+			} catch (CorruptDataException ex) {
+				raiseCorruptDataEvent("CDE thrown extracting initial stack walk state. walkThread = " + walkThread.getHexAddress(), ex, true);
+				return StackWalkResult.STACK_CORRUPT;
+			}
+			return walkStackFrames(walkState);
+		}
+
+		private Object getField(Lookup lookup, Object obj, Class<?> refc, String name, Class<?> rtype) throws Throwable {
+			MethodHandle mh = lookup.findVirtual(refc, name, MethodType.methodType(rtype));
+			return mh.invoke(obj);
+		}
+
+		@Override
+		public StackWalkResult walkStackFrames(WalkState walkState, long continuationAddress) {
 			try {
-				return walkStackFrames(walkState, 
-						walkState.walkThread.sp(), 
-						walkState.walkThread.arg0EA(), 
-						walkState.walkThread.pc(), 
-						walkState.walkThread.literals(),
-						walkState.walkThread.entryLocalStorage());
+				Class<?> contClass = Class.forName("com.ibm.j9ddr.vm29.pointer.generated.J9VMContinuationPointer");
+				Lookup lookup = MethodHandles.lookup();
+				MethodHandle mhCast = lookup.findStatic(contClass, "cast", MethodType.methodType(contClass, long.class));
+				Object cont = mhCast.invoke(continuationAddress);
+
+				walkState.threadAddress = (long)getField(lookup, cont, contClass, "getAddress", long.class);
+				walkState.arg0EA = (UDATAPointer)getField(lookup, cont, contClass, "arg0EA", UDATAPointer.class);
+				walkState.pc = (U8Pointer)getField(lookup, cont, contClass, "pc", U8Pointer.class);
+				walkState.pcAddress = (PointerPointer)getField(lookup, cont, contClass, "pcEA", PointerPointer.class);
+				walkState.walkSP = (UDATAPointer)getField(lookup, cont, contClass, "sp", UDATAPointer.class);
+				walkState.literals  = (J9MethodPointer)getField(lookup, cont, contClass, "literals", J9MethodPointer.class);
+				walkState.privateFlags  = new UDATA(0);
+				walkState.j2iFrame = (UDATAPointer)getField(lookup, cont, contClass, "j2iFrame", UDATAPointer.class);
+				walkState.decompilationStack = (J9JITDecompilationInfoPointer)getField(lookup, cont, contClass, "decompilationStack", J9JITDecompilationInfoPointer.class);
+
+				final J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+				walkState.javaVM = vm;
+				J9VMEntryLocalStoragePointer stubELS = (J9VMEntryLocalStoragePointer)getField(lookup, vm,
```

Comment:
stubELS is not defined as a pointer, you will have to either update the definition in j9nonbuilder to make it a pointer and allocate it in stdinit.c or jclcinit.c. Or get the actual address here.

Reply:
You mean change to `entryLocalStorageEA` instead? I remember `entryLocalStorage` and `entryLocalStorageEA` actually do the same thing if ELS is not a pointer, otherwise I'll have a runtime error.

Reply:
something similar to `jitGPRs`

Reply:
Could you elaborate? I still think `entryLocalStorage()` returns the address of the ELS after went through some code in dtfj. What I did on `jitGPRs` is just one extra casting due to different type in `J9VMContinuation` and in `J9VMEntryLocalStorage`

Reply:
nvm, you are right, I was looking at the generated code for non-struct fields which directly return values, for struct fields `nonNullFieldEA` is used to return the address